examples/llm_inference/ios/InferenceExample/OnDeviceModel.swift

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksGenAI
+
+final class OnDeviceModel {
+
+  private var inference: LlmInference! = {
+    let path = Bundle.main.path(forResource: "model_cpu", ofType: "tflite")!
+    let llmOptions = LlmInference.Options(modelPath: path)
+    return LlmInference(options: llmOptions)
+  }()
+
+  // Temporary to remove extra replace characters left over in model output.
+  // Remove this function once no longer necessary.
+  private func cleanOutput(_ string: String) -> String {
+    let replaceCharacter: Character = "\u{FFFD}"
+    return string.reduce("") { partialResult, next in
+      guard let last = partialResult.last else {
+        if next == replaceCharacter {
+          return partialResult
+        }
+        return String(next)
+      }
+
+      let lastCharacterIsWhitespace = last.unicodeScalars.reduce(false, {
+        $0 || CharacterSet.whitespacesAndNewlines.contains($1)
+      })
+      if next == replaceCharacter {
+        if lastCharacterIsWhitespace {
+          return partialResult
+        }
+        return partialResult + " "
+      }
+      return partialResult + String(next)
+    }
+  }
+
+  func generateResponse(prompt: String) async throws -> String {
+    var partialResult = ""
+    return try await withCheckedThrowingContinuation { continuation in
+      do {
+        try inference.generateResponseAsync(inputText: prompt) { partialResponse, error in
+          if let error = error {
+            continuation.resume(throwing: error)
+            return
+          }
+          if let partial = partialResponse {
+            partialResult += partial.trimmingCharacters(in: .illegalCharacters)
+          }
+        } completion: {
+          let aggregate = self.cleanOutput(partialResult)
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+          continuation.resume(returning: aggregate)
```

Comment:
Since we are not updating the UI on the fly (it seems) we can probably use the method that returns a string here rather than the callback approach.